deck.gl__layers/index.d.ts

```diff
@@ -704,10 +704,173 @@ declare module '@deck.gl/layers/geojson-layer/geojson' {
 }
 declare module '@deck.gl/layers/geojson-layer/geojson-layer' {
 	import { CompositeLayer } from '@deck.gl/core';
-    import { CompositeLayerProps } from "@deck.gl/core/lib/composite-layer";
+	import { LayerProps, LayerData } from "@deck.gl/core/lib/layer";
+	import { CompositeLayerProps } from "@deck.gl/core/lib/composite-layer";
     import PhongMaterial from "@luma.gl/core/materials/phong-material";
     import { RGBAColor } from "@deck.gl/aggregation-layers/utils/color-utils";
-    export interface GeoJsonLayerProps<D> extends CompositeLayerProps<D> {
+
+	// GeoJSON RFC 7946
+	// Project: http://geojson.org/
+	// Definitions derived from: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+	/**
+	 * The valid values for the "type" property of GeoJSON geometry objects.
+	 * https://tools.ietf.org/html/rfc7946#section-1.4
+	 */
+	export type GeoJsonGeometryTypes = Geometry['type'];
+
+	/**
+	 * The value values for the "type" property of GeoJSON Objects.
+	 * https://tools.ietf.org/html/rfc7946#section-1.4
+	 */
+	export type GeoJsonTypes = GeoJSON['type'];
+
+	/**
+	 * Bounding box
+	 * https://tools.ietf.org/html/rfc7946#section-5
+	 */
+	export type BBox = [number, number, number, number] | [number, number, number, number, number, number];
+
+	/**
+	 * A Position is an array of coordinates.
+	 * https://tools.ietf.org/html/rfc7946#section-3.1.1
+	 * Array should contain between two and three elements.
+	 * The previous GeoJSON specification allowed more elements (e.g., which could be used to represent M values),
+	 * but the current specification only allows X, Y, and (optionally) Z to be defined.
+	 */
+	export type Position = [number, number] | [number, number, number];
+
+	/**
+	 * The base GeoJSON object.
+	 * https://tools.ietf.org/html/rfc7946#section-3
+	 * The GeoJSON specification also allows foreign members
+	 * (https://tools.ietf.org/html/rfc7946#section-6.1)
+	 * Developers should use "&" type in TypeScript or extend the interface
+	 * to add these foreign members.
+	 */
+	export interface GeoJsonObject {
+		/**
+		 * Specifies the type of GeoJSON object.
+		 */
+		type: GeoJsonTypes;
+		/**
+		 * Bounding box of the coordinate range of the object's Geometries, Features, or Feature Collections.
+		 * The value of the bbox member is an array of length 2*n where n is the number of dimensions
+		 * represented in the contained geometries, with all axes of the most southwesterly point
+		 * followed by all axes of the more northeasterly point.
+		 * The axes order of a bbox follows the axes order of geometries.
+		 * https://tools.ietf.org/html/rfc7946#section-5
+		 */
+		bbox?: BBox;
+	}
+
+	/**
+	 * Union of GeoJSON objects.
+	 */
+	export type GeoJSON = Geometry | Feature | FeatureCollection;
```

Comment:
@sandcastle Why are we using our own GeoJson definitions? Could we not instead depend on the official [DefinitelyTyped geojson typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/geojson/index.d.ts)? Otherwise we run the issue of being incompatible with this.

Reply:
@MGHawes / @danmarshall - yes we certainly could take a dependency on another project, I just didn't want to be the one to make the call given - it's not my code base. As mentioned in the comments, its a copy/paste from that repo.

Reply:
Thanks for the consideration @sandcastle. Since the GeoJson typings are small (175 lines) I'm ok with taking them as a dependency.